alf/utils/checkpoint_utils.py

```diff
@@ -249,9 +248,9 @@ def _merge_checkpoint(merged, new):
                 "Checkpoint '%s' does not exist. Train from scratch." % f_path)
             return self._global_step
 
-        map_location = None
-        if not torch.cuda.is_available():
-            map_location = torch.device('cpu')
+        # use ``cpu`` as the map location to avoid GPU RAM surge when loading a
+        # model checkpoint.
+        map_location = torch.device('cpu')
```

Comment:
Do we need to convert the model weights to gpu later?

Reply:
> Do we need to convert the model weights to gpu later?

No need to do that. I have been using this since I encountered the out of GPU memory issue when loading checkpoints.

Reply:
Also added a test on this.